src/components/RadioInputGroup/RadioInput/RadioInput.test.tsx

```diff
@@ -0,0 +1,120 @@
+  });
+
+  it('should not be checked if isChecked is not provided', () => {
+    render(<RadioInput {...defaultProps} />);
+    expect(screen.getByRole('radio')).not.toBeChecked();
+  });
+
+  it('should be checked if isChecked is provided', () => {
+    render(<RadioInput {...defaultProps} isChecked />);
+    expect(screen.getByRole('radio')).toBeChecked();
+  });
+
+  describe('User events', () => {
+    it('should call the provided onChange callback on user click', async () => {
+      const user = userEvent.setup();
+      const onChangeMock = jest.fn();
+      render(<RadioInput {...defaultProps} onChange={onChangeMock} />);
+      expect(onChangeMock).not.toHaveBeenCalled();
+
+      await user.click(screen.getByRole('radio'));
+      expect(onChangeMock).toHaveBeenCalled();
+    });
+
+    it('should return the provided id and value from the provided onChange callback on user click', async () => {
+      const user = userEvent.setup();
+      const id = 'test id';
+      const value = 'Test Value';
+      const onChangeMock = jest.fn();
+      render(
+        <RadioInput
+          {...defaultProps}
+          id={id}
+          value={value}
+          onChange={onChangeMock}
+        />,
+      );
+
+      await user.click(screen.getByRole('radio'));
+      expect(onChangeMock).toHaveBeenCalledWith(id, value);
+    });
+
+    it('should not be focusable if it is not checked and isFocusable is not provided', async () => {
+      const user = userEvent.setup();
+      render(<RadioInput {...defaultProps} />);
+      expect(screen.getByRole('radio')).not.toHaveFocus();
+
+      user.tab();
+      expect(screen.getByRole('radio')).not.toHaveFocus();
+    });
+
+    it('should be focusable if it is checked', async () => {
+      const user = userEvent.setup();
+      render(<RadioInput {...defaultProps} isChecked />);
+      expect(screen.getByRole('radio')).not.toHaveFocus();
+
+      user.tab();
+      expect(screen.getByRole('radio')).toHaveFocus();
+    });
+
+    it('should be focusable if isFocusable is provided', async () => {
+      const user = userEvent.setup();
+      render(<RadioInput {...defaultProps} isFocusable />);
+      expect(screen.getByRole('radio')).not.toHaveFocus();
+
+      user.tab();
+      expect(screen.getByRole('radio')).toHaveFocus();
+    });
+
+    it(`should call the provided onChange callback when user clicks ${KeyCodes.ENTER} key`, async () => {
+      const user = userEvent.setup();
+      const onChangeMock = jest.fn();
+      render(
+        <RadioInput {...defaultProps} onChange={onChangeMock} isFocusable />,
+      );
+      expect(onChangeMock).not.toHaveBeenCalled();
+
+      await user.tab();
+      await user.keyboard(`{${KeyCodes.ENTER}}`);
+      expect(onChangeMock).toHaveBeenCalled();
+    });
+
+    it(`should call the provided onChange callback when user clicks ${KeyCodes.SPACE} key`, async () => {
+      const onChangeMock = jest.fn();
+      render(
```

Comment:
We're not **awaiting** anything here. Either `async` should be removed or await should be added. 
Also - how come this test is so different from the `ENTER` test?

Reply:
Good spot about redundant `async`

It is different because React testing library does not provide all of the necessary fields to the generated keyboard event for space if it was written in the same way as for the `ENTER` test. In other words - it doesn't work. And I left that test in that way because that is the modern way it should have been done